### PR TITLE
chore: Set up integ tests on gh actions

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Copied relevant Amplify JS gh actions and workflow to run geo integration tests only - https://github.com/aws-amplify/amplify-js/tree/main/.github/workflows
- POC of tests running successfully - https://github.com/aws-amplify/maplibre-gl-js-amplify/actions/runs/5881898326
- TODO
  - [x] Run analysis tool on workflow - https://github.com/bin3xish477/ghast/tree/main
```
❯ ghast -d .github/workflows
AUXI missing CODEOWNERS file which can provide additional protections for your workflow files.
AUXI missing SECURITY.md file which is crucial for researchers looking to report a finding.
FILE => coverage.yml
WARN check_for_inline_script
WARN check_for_non_github_managed_actions

FILE => publish.yml

FILE => codeql.yml
WARN check_for_non_github_managed_actions

FILE => callable-e2e-test.yml
WARN check_for_upload_download_artifact_action
WARN check_for_inline_script
WARN check_for_non_github_managed_actions

FILE => callable-e2e-tests.yml

Summary
The following Actions failed to pass one or more checks:
 • .github/workflows/coverage.yml
 • .github/workflows/codeql.yml
 • .github/workflows/callable-e2e-test.yml
```
  - [x] Update the environment secret to use a PAT from our amplify-ui-automation account (currently using my personal one only for testing)

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### CODE REVIEW

**Comment by stocaaro** [comment]
Ghast looks really cool. We received feedback from AppSec to do static analysis on a few security areas, which I added as [a JS script](https://github.com/aws-amplify/amplify-js/pull/11638/commits/f692d93aeca7967278fba5202067b54e3daaa503). We might be able to replace this with Ghast, but I think we would need security review for this tool before turning it on.

Don't see anything concerning here. Has this work been through security review yet?

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "maplibre-gl-js-amplify",
-  "version": "2.0.1",
+  "version": "0.0.0-unstable-0e359a8",
```
Comment (stocaaro) [inline package.json:3]: Do you mean to check this in?

Comment (thaddmt) [reply package.json:3]: whoops! nope thanks for the catch

**Comment by thaddmt** [comment]
> Ghast looks really cool. We received feedback from AppSec to do static analysis on a few security areas, which I added as [a JS script](https://github.com/aws-amplify/amplify-js/pull/11638/commits/f692d93aeca7967278fba5202067b54e3daaa503). We might be able to replace this with Ghast, but I think we would need security review for this tool before turning it on.
> 
> Don't see anything concerning here. Has this work been through security review yet?

I actually received a recommendation to use this tool from appsec!

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -8,18 +8,23 @@ on:
...
     name: Publish
     runs-on: ubuntu-latest
+    needs: [e2e]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f
 
       - name: Use Node.js 16 x64
-        uses: actions/setup-node@v2
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c
```
Comment (wlee221) [inline .github/workflows/publish.yml:23]: Nit, but prefer if we add comment that indicates version number, so that it's easier to know when to upgrade.

**Review by reesscot — APPROVED** [review]

**Pull request merged**